modules/ROOT/pages/patterns/shortest-paths.adoc

```diff
@@ -0,0 +1,545 @@
+For example, the following example uses `SHORTEST 1` to return the length of the shortest path between `Worcester Shrub Hill` and `Bromsgrove`:
+
+.Query
+[source, cypher]
+----
+MATCH p = SHORTEST 1 (wos:Station)-[:LINK]-+(bmv:Station)
+WHERE wos.name = "Worcester Shrub Hill" AND bmv.name = "Bromsgrove"
+RETURN length(p) AS result
+----
+
+[TIP]
+Note that this and the following examples on this page use a quantified relationship (as indicated by the postfix quantifier `+`) after the declared relationship pattern to search for paths that include one or more iterations of a pattern.
+For more information, see xref:patterns/reference.adoc#quantified-relationships[Syntax and semantics - quantified relationships].
+
+.Result
+[role="queryresult",options="header,footer",cols="m"]
+|===
+| result
+
+| 2
+
+1+d|Rows: 1
+
+|===
+
+Although the query returned a single result, there are in fact two paths that are tied for shortest:
+
+image::patterns_shortest_tie.svg[width="500",role="middle"]
+
+Because `1` was specified in `SHORTEST`, only one of the paths is returned.
+Which one is returned is non-deterministic.
+
+If instead `SHORTEST 2` is specified, all shortest paths in this example would be returned, and the result would be deterministic:
+
+.Query
+[source, cypher]
+----
+MATCH p = SHORTEST 2 (wos:Station)-[:LINK]-+(bmv:Station)
+WHERE wos.name = "Worcester Shrub Hill" AND bmv.name = "Bromsgrove"
+RETURN [n in nodes(p) | n.name] AS stops
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="m"]
+|===
+| stops
+
+| ["Worcester Shrub Hill", "Droitwich Spa", "Bromsgrove"]
+| ["Worcester Shrub Hill", "Worcestershire Parkway Ll", "Bromsgrove"]
+
+1+d|Rows: 2
+
+|===
+
+Increasing the number of paths will return the next shortest paths.
+Three paths are tied for the second shortest:
+
+image::patterns_second_shortest_paths.svg[width="500",role="middle"]
+
+The following query returns all three of the second shortest paths, along with the two shortest paths:
+
+.Query
+[source, cypher]
+----
+MATCH p = SHORTEST 5 (wos:Station)-[:LINK]-+(bmv:Station)
+WHERE wos.name = "Worcester Shrub Hill" AND bmv.name = "Bromsgrove"
+RETURN [n in nodes(p) | n.name] AS stops
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="m"]
+|===
+
+| stops
+
+| ["Worcester Shrub Hill", "Droitwich Spa", "Bromsgrove"]
+| ["Worcester Shrub Hill", "Worcestershire Parkway Ll", "Bromsgrove"]
+| ["Worcester Shrub Hill", "Worcester Foregate Street", "Droitwich Spa", "Bromsgrove"]
+| ["Worcester Shrub Hill", "Ashchurch", "Worcestershire Parkway Ll", "Bromsgrove"]
+| ["Worcester Shrub Hill", "Ashchurch", "Cheltenham Spa", "Bromsgrove"]
+
+1+d|Rows: 5
+|===
+
+If there had been only four possible paths between the two Stations, then only those four would have been returned.
+
+[[all-shortest]]
+== `ALL SHORTEST`
+
+An alternative way to return all paths that are tied for the shortest length is to use the keywords `ALL SHORTEST`:
```

Comment:
I don't perceive this as "an alternative". _This_ is the way to return all paths that are tied for the shortest length.

Above, we achieve that by writing `SHORTEST 2`, but that obviously is not how to do that in general.

Reply:
Good point. Changed.